test/Microsoft.Health.Fhir.Tests.E2E/Rest/HistoryTests.cs

```diff
@@ -213,11 +213,35 @@ public async Task WhenGettingSystemHistory_GivenAQueryThatReturnsMoreThan10Resul
             newResources.Add(await Client.CreateAsync(Samples.GetJsonSample("ObservationWith1MinuteApgarScore").ToPoco()));
             newResources.Add(await Client.CreateAsync(Samples.GetJsonSample("ObservationWith20MinuteApgarScore").ToPoco()));
 
-            Thread.Sleep(500); // Leave a small gap in the timestamp
+            Thread.Sleep(1000); // Leave a small gap in the timestamp
             var before = DateTime.UtcNow;
 
-            var sinceUriString = HttpUtility.UrlEncode(since.ToString("o"));
+            var sinceUriString = string.Empty;
             var beforeUriString = HttpUtility.UrlEncode(before.ToString("o"));
+            bool historyHas10Entries = false;
+
+            // wait for above values to be represented in history results
+            for (int i = 0; i < 5; i++)
+            {
+                sinceUriString = HttpUtility.UrlEncode(since.ToString("o"));
+                FhirResponse<Bundle> readResponse = Client.SearchAsync("_history?_since=" + sinceUriString).Result;
+
+                if (readResponse.Resource.Entry.Count == 10)
+                {
+                    historyHas10Entries = true;
+                    break;
+                }
+
+                Thread.Sleep(2000);
+                since = DateTime.UtcNow;
```

Comment:
Why is this being reset here? Wouldn't this never work if there was a lag and say one entry was created before we reset it and the other 9 after?

Reply:
That is correct!  Thanks for catching this.